test/test_view_ops.py

```diff
@@ -397,6 +397,9 @@ def test_diagonal_view(self, device) -> None:
         v[0, 0] = 0
         self.assertEqual(t[0, 0, 1], v[0, 0])
 
+        inverse_t = t.diagonal_scatter(v, offset=1, dim1=1, dim2=2)
+        self.assertEqual(inverse_t, t)
```

Comment:
This test doesn't seem to do what you want it to; as `v[0, 0] = 0` mutated the originating tensor `t`, so even if `diagonal_scatter` returned `t` unchanged, this test would pass anyway.

Reply:
I'm going to kill this anyway in favor of the OpInfo tests